File: rails/game/state/BooleanState.java
package rails.game.state;

/**
 * A stateful version of a boolean variable
 * 
 * @author Erik Vos, Stefan Frey (V2.0)
 */
public final class BooleanState extends State {
    
    private boolean value;
    
    private BooleanState(boolean value) {
        this.value = value;
    }
    
    /** 
     * Creates a BooleanState with default value false
     */
    public static BooleanState create(){
        return new BooleanState(false);
    }
    
    /**
     * @param value initial value
     */
    public static BooleanState create(Boolean value){
        return new BooleanState(value);
    }
    
    public void set(boolean value) {
        new BooleanChange(this, value);
    }

    public boolean booleanValue() {
        return value;
    }
    
    @Override
    public String toString() {
        return Boolean.toString(value);
    }

    void change(boolean value) {
        this.value = value;
    }
 
}

File: dotnet/RailsLib.Net/Net/Game/State/BooleanState.cs
using System;

/**
 * A stateful version of a bool variable
 */
namespace GameLib.Net.Game.State
{
    public class BooleanState : GameState
    {
        private bool value;

        private BooleanState(IItem parent, string id, bool value) : base(parent, id)
        {
            this.value = value;
        }

        /** 
         * Creates a BooleanState with default value false
         */
        public static BooleanState Create(IItem parent, string id)
        {
            return new BooleanState(parent, id, false);
        }

        /**
         * Creates a BooleanState with defined initial value
         * @param value initial value
         */
        public static BooleanState Create(IItem parent, string id, Boolean value)
        {
            return new BooleanState(parent, id, value);
        }

        /**
         * @param value set state to this value
         */
        public void Set(bool value)
        {
            if (value != this.value) new BooleanChange(this, value);
        }

        /**
         * @return current value of state variable
         */
        public bool Value
        {
            get
            {
                return value;
            }
        }

        override public string ToText()
        {
            return value.ToString();
        }

        public void Change(bool value)
        {
            this.value = value;
        }
    }
}
